test-resources/headless-core-stub/lambdas/start/src/errors/error-response.ts

```diff
@@ -0,0 +1,21 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { HeadlessCoreStubError } from "./headless-core-stub-error";
+
+export const handleErrorResponse = (err: unknown, logger: Logger) => {
+    if (err instanceof Error) {
+        logger.error(err.message, err);
+    }
+    if (!(err instanceof HeadlessCoreStubError)) {
```

Comment:
I worry that we would need to remember to update this if we started adding new error types. What do you think about sometihng like just 
```
const message = err.message || "Server error";
return {
    statusCode: err.status || 500,
    body: JSON.stringify({ message }),
}
```

I haven't seen this idea of using different error types to control error handling in TypeScript so much, but if you've got some docs/prior art please share!